doc/modules/impute.rst

```diff
@@ -219,10 +219,39 @@ neighbors of samples with missing values::
            [5.5, 6. , 5. ],
            [8. , 8. , 7. ]])
 
-.. [OL2001] Olga Troyanskaya, Michael Cantor, Gavin Sherlock, Pat Brown,
-    Trevor Hastie, Robert Tibshirani, David Botstein and Russ B. Altman,
-    Missing value estimation methods for DNA microarrays, BIOINFORMATICS
-    Vol. 17 no. 6, 2001 Pages 520-525.
+.. topic:: References
+
+  .. [OL2001] Olga Troyanskaya, Michael Cantor, Gavin Sherlock, Pat Brown,
+      Trevor Hastie, Robert Tibshirani, David Botstein and Russ B. Altman,
+      Missing value estimation methods for DNA microarrays, BIOINFORMATICS
+      Vol. 17 no. 6, 2001 Pages 520-525.
+
+Keeping the number of features constants
+========================================
+
+By default, the scikit-learn imputers will drop fully empty features, i.e.
+columns containing only missing values. For instance::
+
+  >>> imputer = SimpleImputer()
+  >>> X = np.array([[np.nan, 1], [np.nan, 2], [np.nan, 3]])
+  >>> imputer.fit_transform(X)
+  array([[1.],
+         [2.],
+         [3.]])
+
+The first feature in `X` containing only `np.nan` was dropped after the
+imputation. While this feature will not help in predictive setting, dropping
+the columns, will change the shape of `X` which could be problematic when using
+imputers in a more complex machine-learning pipeline. The parameter
+`keep_empty_features` offer the option to keep the empty features by imputing
+with a constant values. In most of the cases, this constant value is zero::
```

Comment:
Thanks for the documentation of this behavior.

Should we add a remark and a doc-test as an example using `strategy="constant"` and `fill_value`?

Reply:
I would let this specific implementation details be documented only in the API documentation.